lib/credentials/shared_ini_file_credentials.js

```diff
@@ -87,6 +100,75 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
   /**
    * @api private
    */
+  loadRoleProfile: function loadRoleProfile(creds, callback) {
```

Comment:
We could probably just pass the `roleProfile` instead of `creds` so we don't have to grab the `roleProfile` later. I don't think we'll have to worry about having the whole creds file in the future for this method.